akka-actor-tests/src/test/scala/akka/io/TcpConnectionSpec.scala

```diff
@@ -232,23 +233,25 @@ class TcpConnectionSpec extends AkkaSpec("""
 
     "write file to network" in new EstablishedConnectionTest() {
       run {
-        // hacky: we need a file for testing purposes, so try to get the biggest one from our own classpath
-        val testFile =
-          classOf[TcpConnectionSpec].getClassLoader.asInstanceOf[URLClassLoader]
-            .getURLs
-            .filter(_.getProtocol == "file")
-            .map(url ⇒ new File(url.toURI))
-            .filter(_.exists)
-            .sortBy(-_.length)
-            .head
-
-        // maximum of 100 MB
-        val size = math.min(testFile.length(), 100000000).toInt
-
-        val writer = TestProbe()
-        writer.send(connectionActor, WriteFile(testFile.getAbsolutePath, 0, size, Ack))
-        pullFromServerSide(size, 1000000)
-        writer.expectMsg(Ack)
+        val tmpFile = Files.createTempFile("akka-io-spec-write-file-to-network", ".dat")
```

Comment:
-1, please use jimfs so this does not become a fleaky test

Reply:
See FileSinkSpec to learn how:

```scala
val fs = Jimfs.newFileSystem("FileSinkSpec", Configuration.unix())

  val TestLines = {
    val b = ListBuffer[String]()
    b.append("a" * 1000 + "\n")
    b.append("b" * 1000 + "\n")
    b.append("c" * 1000 + "\n")
    b.append("d" * 1000 + "\n")
    b.append("e" * 1000 + "\n")
    b.append("f" * 1000 + "\n")
    b.toList
  }
```

Reply:
You sure the io-infra handles inmem filesystem, I thought the WriteFile boiled down to some os support for transferring files?

Reply:
(That's the only reason I went with this)

Reply:
Ah, I see now I was wrong, `FileChannel.open` etc. so we could use paths all the way indeed.

Reply:
yeah all the APIs to use it with everything do exist, mostly a matter of our message using a string there it seems?